.github/actions/build-avr/action.yml

```diff
@@ -0,0 +1,6 @@
+name: 'Build examples'
+author: 'Matthew McGowan'
+description: 'Build the library examples, targeting various platforms'
```

Comment:
I see we also have code targeting the NRF52, ESP32 etc.. which are beyond the AVR, hence I was asking for a list of platforms that we will spend effort in supporting.

Reply:
ESP32 and STM32 are our primary targets, but we should also be checking NRF52, AVR, SAMD and MBED (RP2040).

Reply:
Also, I guess I'm hung up on the fact the action is labelled `build-avr` and not `build-examples`.

I like that each platform has it's own build action, because build actions run in parallel.